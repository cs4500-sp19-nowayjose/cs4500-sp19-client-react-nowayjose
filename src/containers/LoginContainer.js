import React from 'react'
import UserService from '../services/UserService'
import Login from '../components/Login/Login'

class LoginContainer extends React.Component {
    constructor(props) {
        super(props);

        this.UserService = UserService.getInstance()

        this.state = {
            username: "",
            password: ""
        }

        this.handleChange = this.handleChange.bind(this)
        this.handleLogin = this.handleLogin.bind(this)
    }

    handleChange = event =>  {
        this.setState({
            [event.target.id]: event.target.value
        })
    }

    handleLogin = event => {
        var credentials = {
            username: this.state.username,
            password: this.state.password
        }

        this.UserService.loginUser(credentials).then(resp => {
            if (resp.status === 403) {
                alert('Invalid credentials. Please try again.')
            }
            else {
                this.props.history.push('/profile')
            }
        })
    }

    render() {
        return (<Login 
            handleChange = {this.handleChange}
            handleLogin = {this.handleLogin}
        />)
    }
}

export default LoginContainer