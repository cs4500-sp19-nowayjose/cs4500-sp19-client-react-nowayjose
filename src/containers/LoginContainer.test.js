import React from 'react'
import LoginContainer from './LoginContainer'
import TestRenderer from 'react-test-renderer'

test('[LoginContainer renders correctly]', () => {
    const testRenderer = TestRenderer.create(<LoginContainer />)
    let tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
})

test('[LoginContainer renders correctly after hitting login]', () => {
    const testRenderer = TestRenderer.create(<LoginContainer />)
    let tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    const testInstance = testRenderer.root
    let loginButton = testInstance.findByProps({id: 'loginBtn'})
    loginButton.props.onClick({})

    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
})