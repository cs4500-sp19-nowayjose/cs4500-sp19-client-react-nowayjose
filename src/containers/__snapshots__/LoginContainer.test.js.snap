// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[LoginContainer renders correctly after hitting login] 1`] = `
<div
  className="container"
>
  <h1>
    Welcome back
  </h1>
  <br />
  <div>
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <label
          for="username"
        >
          Username
        </label>
        <input
          className="form-control"
          id="username"
          onChange={[Function]}
        />
      </div>
      <div
        className="col-12"
      >
        <br />
        <label
          for="password"
        >
          Password
        </label>
        <input
          className="form-control"
          id="password"
          onChange={[Function]}
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-6"
      >
        <label
          for="password"
        >
          <input
            type="checkbox"
          />
            Remember me
        </label>
      </div>
      <div
        className="col-6"
      >
        <a
          className="float-right"
          href="#"
        >
          Forgot password?
        </a>
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <button
          className="btn btn-primary btn-block"
          id="loginBtn"
          onClick={[Function]}
        >
          Log in
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`[LoginContainer renders correctly after hitting login] 2`] = `
<div
  className="container"
>
  <h1>
    Welcome back
  </h1>
  <br />
  <div>
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <label
          for="username"
        >
          Username
        </label>
        <input
          className="form-control"
          id="username"
          onChange={[Function]}
        />
      </div>
      <div
        className="col-12"
      >
        <br />
        <label
          for="password"
        >
          Password
        </label>
        <input
          className="form-control"
          id="password"
          onChange={[Function]}
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-6"
      >
        <label
          for="password"
        >
          <input
            type="checkbox"
          />
            Remember me
        </label>
      </div>
      <div
        className="col-6"
      >
        <a
          className="float-right"
          href="#"
        >
          Forgot password?
        </a>
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <button
          className="btn btn-primary btn-block"
          id="loginBtn"
          onClick={[Function]}
        >
          Log in
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`[LoginContainer renders correctly] 1`] = `
<div
  className="container"
>
  <h1>
    Welcome back
  </h1>
  <br />
  <div>
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <label
          for="username"
        >
          Username
        </label>
        <input
          className="form-control"
          id="username"
          onChange={[Function]}
        />
      </div>
      <div
        className="col-12"
      >
        <br />
        <label
          for="password"
        >
          Password
        </label>
        <input
          className="form-control"
          id="password"
          onChange={[Function]}
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-6"
      >
        <label
          for="password"
        >
          <input
            type="checkbox"
          />
            Remember me
        </label>
      </div>
      <div
        className="col-6"
      >
        <a
          className="float-right"
          href="#"
        >
          Forgot password?
        </a>
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <button
          className="btn btn-primary btn-block"
          id="loginBtn"
          onClick={[Function]}
        >
          Log in
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`[LoginContainer renders correctly] 2`] = `
<div
  className="container"
>
  <h1>
    Welcome back
  </h1>
  <br />
  <div>
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <label
          for="username"
        >
          Username
        </label>
        <input
          className="form-control"
          id="username"
          onChange={[Function]}
        />
      </div>
      <div
        className="col-12"
      >
        <br />
        <label
          for="password"
        >
          Password
        </label>
        <input
          className="form-control"
          id="password"
          onChange={[Function]}
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-6"
      >
        <label
          for="password"
        >
          <input
            type="checkbox"
          />
            Remember me
        </label>
      </div>
      <div
        className="col-6"
      >
        <a
          className="float-right"
          href="#"
        >
          Forgot password?
        </a>
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <button
          className="btn btn-primary btn-block"
          id="loginBtn"
          onClick={[Function]}
        >
          Log in
        </button>
      </div>
    </div>
  </div>
</div>
`;
